core/lib/Cardano/Address/Errors.hs

```diff
@@ -0,0 +1,42 @@
+{-# LANGUAGE ExistentialQuantification #-}
+
+-- |
+-- Copyright: © 2020 IOHK
+-- License: Apache-2.0
+
+module Cardano.Address.Errors
+    ( -- $overview
+
+      -- * AddrError
+      AddrError(..)
+  ) where
+
+import Prelude
+
+import Cardano.Address
+    ( Address )
+
+import Control.Exception
+    ( Exception )
+
+-- $overview
+--
+-- This module provides error/exception types for Addresses.
+
```

Comment:
I don't think we can get away with a single type for all `inspectXXX` functions in this module. For example, Jormungandr addresses will never throw an `InvalidDerivationPath` error and, "pure Shelley" addresses won't either. So it may be better to distinguish between Shelley and Byron errors (and Jormungandr on a third type).

Reply:
Sure, since it's MonadThrow, that's easily possible. My initial idea was just for the cli to print better errors. If someone wants to depend on the constructors for control flow, then I guess we need to distinguish.